great_expectations_cloud/agent/actions/draft_datasource_config_action.py

```diff
@@ -58,7 +63,42 @@ def check_draft_datasource_config(
         datasource = datasource_cls(**draft_config)
         datasource._data_context = self._context
         datasource.test_connection(test_assets=True)  # raises `TestConnectionError` on failure
-        return ActionResult(id=id, type=event.type, created_resources=[])
+
+        if isinstance(datasource, SQLDatasource):
+            table_names = self._get_table_names(datasource=datasource)
+            self._update_table_names_list(config_id=event.config_id, table_names=table_names)
+
+        return ActionResult(
+            id=id,
+            type=event.type,
+            created_resources=[],
+        )
+
+    def _get_table_names(self, datasource: Datasource) -> list[str]:
+        inspector: Inspector = inspect(datasource.get_engine())
+        return inspector.get_table_names()
+
+    def _update_table_names_list(self, config_id: UUID, table_names: list[str]) -> None:
+        try:
+            cloud_config = GxAgentEnvVars()
+        except pydantic.ValidationError as validation_err:
+            raise RuntimeError(
+                generate_config_validation_error_text(validation_err)
+            ) from validation_err
+
+        session = create_session(access_token=cloud_config.gx_cloud_access_token)
+        response = session.patch(
+            url=f"{cloud_config.gx_cloud_base_url}/organizations/"
+            f"{cloud_config.gx_cloud_organization_id}/datasources/drafts/{config_id}",
+            json={"table_names": table_names},
+        )
+        if not response.ok:
+            raise RuntimeError(  # noqa: TRY003 # one off error
+                f"DraftDatasourceConfigAction encountered an error while connecting to GX Cloud. "
+                f"Unable to update "
+                f"table_names for Draft Config with ID"
+                f"={config_id}.",
+            )
```

Comment:
`non-blocking`
Do all of the agent jobs raise `RuntimeError`s?

Reply:
Yep